### PR TITLE
BFCL April 10th Release 

### PR DESCRIPTION
**Pull request by HuanzhiMao**
This PR is for the BFCL April 10th release:

1. Introduce REST API sanity check for the executable test category. It ensures that all the API endpoints involved during the execution evaluation process are working properly. If any of them are not behaving as expected, the evaluation process will be stopped by default as the result will be inaccurate. Users can choose to bypass this check by setting the `--skip-api-sanity-check` flag.

This PR **DOES NOT** change the leaderboard score.

### CODE REVIEW

File: berkeley-function-call-leaderboard/eval_checker/eval_runner_helper.py [hunk 1]
```diff
@@ -32,6 +37,7 @@
     "Latency Standard Deviation (s)",
 ]
 
+# Note that we don't need to
```
Comment (CharlieJCJ) [inline berkeley-function-call-leaderboard/eval_checker/eval_runner_helper.py:40]: Write a more elaborate comment here

File: berkeley-function-call-leaderboard/eval_checker/eval_runner_helper.py [hunk 2]
```diff
@@ -717,7 +752,6 @@ def update_leaderboard_table_with_score_file(leaderboard_table, score_path):
 
 def oss_file_formatter(input_file_path, output_dir):
     data = load_file(input_file_path)
-    assert len(data) == 2000, "OSS result.json file should have 2000 entries."
```
Comment (CharlieJCJ) [inline berkeley-function-call-leaderboard/eval_checker/eval_runner_helper.py:720]: Revert back

File: berkeley-function-call-leaderboard/eval_checker/eval_runner.py [hunk 3]
```diff
@@ -284,11 +290,12 @@ def runner(model_names, test_categories):
         if len(files) == 1 and files[0] == "result.json":
             result_json_file_path = os.path.join(subdir, "result.json")
             oss_file_formatter(result_json_file_path, subdir)
-            print(f"Detected OSS model: {model_name}. result.json has been split into individual test category files.")
```
[the remark this answers is not in the document]
Comment (HuanzhiMao) [reply berkeley-function-call-leaderboard/eval_checker/eval_runner.py:287]: synced with the latest remote main

File: berkeley-function-call-leaderboard/eval_checker/eval_runner_helper.py (hunk 2 above)
Comment (HuanzhiMao) [reply berkeley-function-call-leaderboard/eval_checker/eval_runner_helper.py:720]: synced with the latest remote main

**Review by CharlieJCJ — APPROVED** [review]
Tested, LGTM

**Review by ShishirPatil — APPROVED** [review]

**Pull request merged**